### ISSUE TITLE
Integrate AI Model for Video Generation via HuggingFace API 

### ISSUE THREAD

**Issue by Manav-Khandurie** [upes-open/OSoc-24-The-Content-Forge#5]
Use the HuggingFace API to run open source Video Models and integrate them to the Image/Video Generation Microservice

Ensure that there is  proper video generation with the model that was chosen , implement the model with lowest running cost possible

Ensure that the video generated can be of 15-30s long and provide the functionality of goin Text --> Video.

The generated videos must be stored in the S3 Bucket , whose credentials will be provided .

**Comment by nandiniiiiiiii** [comment]
sir in this we have to create a model that takes text and convert it to video right....?? using hugging face api and implement it via react??

**Comment by Manav-Khandurie** [comment]
> sir in this we have to create a model that takes text and convert it to video right....?? using hugging face api and implement it via react??

Yes @nandiniiiiiiii  you are almost correct so in this issue basically we need to integrate some text to video from hugging face via its api , but this is done on the backend microservice ,so we have a node/python microservice that accepts an api request with text init and the service communicate with hugging face api to run any free model on hugging face platform and then genrate and send it back .

We can store the image/video generated in a s3 buckets as well

**Comment by nandiniiiiiiii** [comment]
Oooo ok, actually I have worked with hugging face and implemented it using Flask so I thought it was something similar, but it looks a little more complex thankyou so much for clarifying it.

**Comment by Manav-Khandurie** [comment]
> Oooo ok, actually I have worked with hugging face and implemented it using Flask so I thought it was something similar, but it looks a little more complex thankyou so much for clarifying it.

No worries, if you change your mind just let me know

**Comment by adx04** [comment]
> gingFace API to run open source Video Models and integrate them to the Image/Video Generation Microservice
> 
> Ensure that there is proper video generation with the model that was chosen , implement the model with lowest running cost possible
> 
> Ensure that the video generated can be of 15-30s long and provide the functionality of goin Text --> Video.
> 
> The generated videos must be stored in the S3 Bucket , whose credentials will be provided .

in which folder i have to work on?

**Comment by adx04** [comment]
> hello there @Manav-Khandurie ,i want on work on this issue this one looks pretty interesting..can you assign it to me?

in which folder i have to work..pls guide me a little?

**Comment by adx04** [comment]
yes @Manav-Khandurie 


> > hello there @Manav-Khandurie ,i want on work on this issue this one looks pretty interesting..can you assign it to me?
> 
> Are you still interested @adx04 sorry for the late response

Yes @Manav-Khandurie !

**Comment by adx04** [comment]
> > hello there @Manav-Khandurie ,i want on work on this issue this one looks pretty interesting..can you assign it to me?
> 
> Are you still interested @adx04 sorry for the late response

yes @Manav-Khandurie ! pls guide me a liittle on this issue like about frameworks and credentials for the s3 and HuggingFace API credentials

**Comment by rampantvoid** [comment]
@Manav-Khandurie hi can I be assigned on this?

**Comment by Manav-Khandurie** [comment]
> @Manav-Khandurie hi can I be assigned on this?

ok

**Comment by Manav-Khandurie** [comment]
> > > hello there @Manav-Khandurie ,i want on work on this issue this one looks pretty interesting..can you assign it to me?
> > 
> > 
> > Are you still interested @adx04 sorry for the late response
> 
> yes @Manav-Khandurie ! pls guide me a liittle on this issue like about frameworks and credentials for the s3 ?

Sorry for the late response , @adx04  , actually you would be integrating the models form your ac count and just setting up the .env variables with hugging face , s3 upload functionality is done  so leave that aspect now

**Comment by rampantvoid** [comment]
@Manav-Khandurie hey I was working on this and it seems we cannot run a text-to-video or any diffuser model off the hugging face api, they're not yet supported. Already have pinged you regarding this on discord
